Lib/argparse.py

```diff
@@ -1469,10 +1469,8 @@ def _get_optional_kwargs(self, *args, **kwargs):
 
             # strings starting with two prefix characters are long options
             option_strings.append(option_string)
-            if option_string[0] in self.prefix_chars:
-                if len(option_string) > 1:
-                    if option_string[1] in self.prefix_chars:
```

Comment:
I don't think these checks are redundant.

Consider `self.prefix_chars = {'-'}`

The left hand side requires `'--foo'` whereas your updated code incorrectly allows `X-foo`

I do think these conditions could be combined, perhaps something like:

```python
if (
        option_string[0] in self.prefix_chars and
        len(option_string) > 1 and
        option_string[1] in self.prefix_chars
):
    ...
```

Reply:
Please check the code from line no 1463. It checks for the condition `option_string[0] in self.prefix_chars`.

Reply:
Ah indeed! Maybe use `elif` so that's more clear at a glance?

Reply:
@asottile This change looks good. No need for `else`. This looks cleaner.